### DISCUSSION TITLE
Easy access to CORDEX datasets

### DISCUSSION THREAD

**Discussion by larsbuntemeyer** [WCRP-CORDEX/discuss#5] [Ideas]
### Description

(copied this issue here so i can close https://github.com/euro-cordex/py-cordex/issues/39)

To make research reproducible, easy access to input data under FAIR conditions is critical. For EURO-CORDEX (ca. 75TB), it's not trivial if you want to [work with the whole ensemble](https://nbviewer.org/github/euro-cordex/py-cordex/blob/master/notebooks/euro-cordex-ensemble.ipynb) if you have no direct access to the EURO-CORDEX ensemble dataset. Bringing EURO-CORDEX data to a cloud provider would solve lots problems concerning:

* reproducibility
* reusability
* exchange of hands on tutorials with real input data
* avoid selection bias if you have to decide on datasets at download time
* promote the datasets more easily and democratically (low-threshold)

### Possible cloud providers

* [Pangeo Forge](https://pangeo-for.ge/) (under development)
  * https://github.com/pangeo-forge/staged-recipes/issues/96
* [Amazon Open Data Sponsorhip](https://aws.amazon.com/opendata/open-data-sponsorship-program/)
  * e.g., [NA-CORDEX](https://registry.opendata.aws/ncar-na-cordex/)
* [EOSC](https://eosc-portal.eu/): European Open Science Cloud
* [2i2c](https://2i2c.org/)
* [DKRZ S3 MinIO](https://docs.dkrz.de/doc/datastorage/minio/index.html)
* [Pangeo@EOSC](https://www.egi.eu/magazine/issue-2024-02/pangeoeosc-enabling-open-reproducible-and-scalable-big-data-on-the-eosc)

This is also interesting for the [Topic 4](https://www.exaesm.de/topics/topic4/) in Pilot Lab Exascale Earth System Modelling.

### Individual model restrictions

* https://is-enes-data.github.io/CORDEX_RCMs_ToU.txt
* https://github.com/IS-ENES-Data/cordex/blob/master/CORDEX_register.xlsx

**Comment by larsbuntemeyer** [comment]
see also:
https://gitlab.dkrz.de/data-infrastructure-services/zarr-in-swift-objectstorage

**Comment by larsbuntemeyer** [comment]
Development of ESGF extractions is now done here: https://github.com/jbusecke/pangeo-forge-esgf

**Comment by larsbuntemeyer** [comment]
@gnikulin @jesusff

### thoughts on licenses

One more thought on licensing issues: It would be nice if CORDEX datasets (CMIP5) could be made available for unrestricted use without authorization in the future similar to the [2020 update of the CMIP5 terms of use](https://pcmdi.llnl.gov/mips/cmip5/terms-of-use.html). Since the [CORDEX terms of use](https://cds.climate.copernicus.eu/api/v2/terms/static/cordex-licence.pdf) seemed to be inherited from CMIP5, the update might be natural. This should be at least technically and legally possible for [datasets with unrestricted use](https://is-enes-data.github.io/CORDEX_RCMs_ToU.txt). The authorization adds an [unneccessary overhead of complexity](https://github.com/ESGF/esgf-pyclient/issues?q=opendap) for users that wish to access data automatically, e.g.,

* https://github.com/ESGF/esgf-pyclient/issues/84
* https://github.com/ESGF/esgf-pyclient/issues/69
* https://github.com/ESGF/esgf-pyclient/issues/52
* https://github.com/ESGF/esgf-pyclient/issues/26
* https://github.com/ESGF/esgf-pyclient/issues/83
* https://github.com/ESGF/esgf-pyclient/issues/78
* https://github.com/ESGF/esgf-pyclient/issues/55
* etc...

I also see it this as a conflict between the desire to make CORDEX data easily accessible for users but having authorization that prevent automatic use. Especially if you want users to work with the whole ensemble, redistribution to, e.g., zarr formats could be easier...

**Reply by gnikulin** [reply]
In CORDEX we still have a number of institutions with the non-commercial ToU in contrast to CMIP5 where all institutions with the non-commercial ToU have agreed to relax their ToU to the unrestricted ones. We cannot simply remove the authorization for all CORDEX simulations. I don't know if  it's technically possible to remove the authorization for the datasets with the unrestricted ToU. Even in these case users have to be authorised to download the non-commercial datasets to use full ensembles. In CORDEX-CMIP6 we definitely need to skip the authorisation and the ToU will be shown in the global attribute license similar to CMIP6.

**Reply by larsbuntemeyer** [reply]
Thanks @gnikulin ! Maybe, @agstephens could help out (any feedback would be appreciated very much!)? I am basically just throwing around ideas from my head, the question would be, if it is *technically* possible to remove authorization similar to CMIP5 and CMIP6 datasets in ESGF for datasets with an unrestricted CORDEX license. I guess there has to be some internal system, e.g., i can, at least, apply for `CORDEX_Research` and `CORDEX_Commercial` logins.... So there should be some list of whatsoever that distinguishes datasets based on their license...

**Reply by larsbuntemeyer** [reply]
More more thought: it would also really be helpful simply for binder based tutorial notebooks to be able to access, e.g., ESGF CORDEX opendat datasets also for users whithout openid.. This would be nice to create, e.g., Jupyter books without having to store credentials in github secrets, etc..

**Comment by larsbuntemeyer** [comment]
* pangeo-data/pangeo-cmip6-cloud/issues/31
